tests/test_packages/test_connections/test_p2p_libp2p/test_public_dht.py

```diff
@@ -85,171 +78,71 @@ def delegate_uris_public_keys(request):
 class TestLibp2pConnectionPublicDHTRelay(BaseP2PLibp2pTest):
     """Test that public DHT's relay service is working properly"""
 
-    @pytest.mark.parametrize("maddrs", [PUBLIC_DHT_MADDRS], indirect=True)
-    def test_connectivity(self, maddrs):
-        """Test connectivity."""
-
-        for maddr in maddrs:
-            connection = _make_libp2p_connection(relay=False, entry_peers=[maddr])
-            multiplexer = Multiplexer([connection])
-            self.multiplexers.append(multiplexer)
-            self.log_files.append(connection.node.log_file)
-            multiplexer.connect()
-            assert_msg = f"Couldn't connect to public node {maddr}"
-            assert connection.is_connected is True, assert_msg
+    maddrs = PUBLIC_DHT_MADDRS
 
-    @pytest.mark.parametrize("maddrs", [PUBLIC_DHT_MADDRS], indirect=True)
-    def test_communication_direct(self, maddrs):
-        """Test direct communication via each of the multiaddrs."""
-
-        for maddr in maddrs:
-
-            connection1 = _make_libp2p_connection(relay=False, entry_peers=[maddr])
-            multiplexer1 = Multiplexer([connection1])
-            self.log_files.append(connection1.node.log_file)
-            multiplexer1.connect()
-            self.multiplexers.append(multiplexer1)
+    def setup(self):
+        """Setup test"""
+        assert len(self.maddrs) > 1, "Test requires at least 2 public DHT node"
+        for maddr in self.maddrs:
+            for _ in range(2):  # make pairs
+                self.make_connection(relay=False, entry_peers=[maddr])
 
-            connection2 = _make_libp2p_connection(relay=False, entry_peers=[maddr])
-            multiplexer2 = Multiplexer([connection2])
-            self.log_files.append(connection2.node.log_file)
-            multiplexer2.connect()
-            self.multiplexers.append(multiplexer2)
+    def teardown(self):
+        """Teardown after test method"""
+        self._disconnect()
+        self.multiplexers.clear()
+        self.log_files.clear()
+
+    @property
+    def pairs_with_same_entry_peers(self):
+        """Multiplexer pairs with different entry peers"""
+        return itertools.zip_longest(*[iter(self.multiplexers)] * 2)
+
+    @property
+    def pairs_with_different_entry_peers(self):
+        """Multiplexer pairs with different entry peers"""
+        return itertools.permutations(self.multiplexers[::2], 2)
```

Comment:
setup and teardown for each test method, not the class. I expect that interference of the two tests iterating the same set of multiplexers was causing the issue (in tests directly below)